pkg/lang/ir/v1/julia.go

```diff
@@ -16,46 +16,84 @@ package v1
 
 import (
 	"fmt"
+	"path/filepath"
 	"strings"
 
-	"github.com/cockroachdb/errors"
 	"github.com/moby/buildkit/client/llb"
-	"github.com/sirupsen/logrus"
+)
 
-	"github.com/tensorchord/envd/pkg/types"
+const (
+	juliaRootDir = "/opt/julia"                             // Location of downloaded Julia binary and other files
+	juliaBinDir  = "/opt/julia/bin"                         // Location of Julia executable binary file
+	juliaPkgDir  = "/opt/julia/user_packages"               // Location of additional packages installed via Julia
+	juliaBaseURL = "https://julialang-s3.julialang.org/bin" // The official link for downloading Julia environment
+	juliaOS      = "linux"                                  // Platform operating system
+	juliaArch    = "x64"                                    // Platform architecture
+	juliaVersion = "1.8"                                    // Julia version
+	juliaBinName = "julia-1.8.3-linux-x86_64.tar.gz"        // Julia archive name
 )
 
-func (g *generalGraph) installJulia(root llb.State) (llb.State, error) {
-	root = g.updateEnvPath(root, types.DefaultJuliaPath)
-	return root, errors.New("not implemented")
+// getJuliaBinary returns the llb.State only after setting up Julia environment
+// A successful run of getJuliaBinary should set up the Julia environment
+func (g generalGraph) getJuliaBinary(root llb.State) llb.State {
+
+	var juliaDownloadURL = filepath.Join(juliaBaseURL, juliaOS, juliaArch, juliaVersion, juliaBinName)
```

Comment:
Cannot use `filepath.Join` on URL. BTW, the juliaBinName also contains the version info.

Reply:
Julia download website consider `1.8.1` ,`1.8.2` etc all to `1.8` category. Similarly for `1.7`, `1.6`. If we want to construct the download link with just the archive file name, we need to parse its name to calculate the category. I think that would be complex

Reply:
It reminds me that the current implementation doesn't support ARM.